packages/flame/lib/src/components/mixins/has_paint.dart

```diff
@@ -97,4 +98,32 @@ mixin HasPaint<T extends Object> on Component {
   void tint(Color color, {T? paintId}) {
     getPaint(paintId).colorFilter = ColorFilter.mode(color, BlendMode.srcATop);
   }
+
+  @override
+  double get opacity => paint.color.opacity;
+
+  @override
+  set opacity(double value) {
+    paint.color = paint.color.withOpacity(value);
+    for (final paint in _paints.values) {
+      paint.color = paint.color.withOpacity(value);
+    }
+  }
+
+  OpacityProvider opacityProviderForPaint(T paintId) {
```

Comment:
Maybe call it `opacityOfPaint(paintId)` for short?

`OpacityEffect(target: component.opacityOfPaint('bg'))` sounds nicer.

Reply:
It sounds like you'd get back the `double` from `opacityOfPaint` though?

Reply:
Well, you'd get something that has `.opacity` property; it's a bit like getting an `Opacity` object instead of a simple value. And there is definitely no confusion when used as a target of an `OpacityEffect`.

Reply:
There is no confusion when used as a target, but the user will get this method suggested when doing other things on the component too. Since we have no opacity object I don't think any user would expect that.

Reply:
A short name would have been better, but just like @spydon said, `opacityOfPaint` will be a misnomer. How about `opacityProviderOf`? The input parameter already suggests that it operates on paint